include/dlaf/tile.h

```diff
@@ -53,19 +59,28 @@ class Tile {
   template <class U = T,
             class = typename std::enable_if_t<std::is_const<U>::value && std::is_same<T, U>::value>>
   Tile(Tile<ElementType, device>&& rhs)
-      : m_(rhs.m_), n_(rhs.n_), memory_view_(std::move(rhs.memory_view_)), ld_(rhs.ld_) {
+      : m_(rhs.m_), n_(rhs.n_), memory_view_(std::move(rhs.memory_view_)), ld_(rhs.ld_),
+        p_(std::move(rhs.p_)) {
     rhs.m_ = 0;
     rhs.n_ = 0;
     rhs.ld_ = 1;
   }
 
+  ~Tile() {
+    if (p_) {
+      auto p = std::move(p_);
```

Comment:
Is it really needed to move it?

In my knowledge it is not and, in case, I would remove it (IMHO it does not improve readability and moreover we can "save" a couple of lines).

Reply:
Not anymore, can be removed.